subsys/net/lib/coap/Kconfig

```diff
@@ -152,6 +152,11 @@ config COAP_CLIENT_MAX_REQUESTS
 	help
 	  Maximum number of CoAP requests a single client can handle at a time
 
+config COAP_CLIENT_TRUNCATE_MSGS
+	bool "Truncate blocks larger than the buffer size"
```

Comment:
I think this should be `y` by default so that the system behaves the same way as before the change.

Reply:
I considered that. I argue that the PR that added this inadvertently broke all reading operations for previously working platforms, so this PR just returns it to the previous behavior socket-wise.

Reply:
Perhaps some something like:
```
default n if <Kconfig for specific offloaded network stack that doesn't support this option>
default y
```
might be a good compromise?

Reply:
Also I don't think the description of the Kconfig option is quite right at the moment - the packet is always truncated if larger than the receive buffer, passing the `MSG_TRUNC` flag just means that we are informed the packet has been truncated rather than this happening silently.

Reply:
> I argue that the PR that added this inadvertently broke all reading operations for previously working platforms, so this PR just returns it to the previous behavior socket-wise.

IMO we should not be limiting app libraries functionalities only because some offloaded implementations does not provide features supported by the native stack. While it's ok to have this covered by a Kconfig option, IMO it should be enabled by default upstream. We can override the default in NCS if modemlib is selected.

> Also I don't think the description of the Kconfig option is quite right at the moment - the packet is always truncated if larger than the receive buffer, passing the MSG_TRUNC flag just means that we are informed the packet has been truncated rather than this happening silently.

Agreed, the truncation happens regardless, MSG_TRUNC just gives info about it.

Reply:
OK, made these requested changes.